pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -67,7 +69,19 @@ func (e ShortcutExpander) getAll() []schema.GroupResource {
 		}
 	}
 
-	return availableAll
+	// populate all "discovered" resources from the server
+	for availableResource := range availableResources {
```

Comment:
Hm, this just became a bit more tricky. Other than maybe specifying the contents of `UserResources` as part of a client config, I am not sure how else to handle cases where none of the resources in `UserResources` match any of the discovered ones.